srtcore/epoll.cpp

```diff
@@ -125,10 +125,66 @@ ENOMEM: There was insufficient memory to create the kernel object.
    pair<map<int, CEPollDesc>::iterator, bool> res = m_mPolls.insert(make_pair(m_iIDSeed, CEPollDesc(m_iIDSeed, localid)));
    if (!res.second)  // Insertion failed (no memory?)
        throw CUDTException(MJ_SETUP, MN_NONE);
+   if (pout)
+       *pout = &res.first->second;
 
    return m_iIDSeed;
 }
 
+int CEPoll::clear_usocks(int eid)
+{
+    // This should remove all SRT sockets from given eid.
+   CGuard pg (m_EPollLock);
+
+   map<int, CEPollDesc>::iterator p = m_mPolls.find(eid);
+   if (p == m_mPolls.end())
+      throw CUDTException(MJ_NOTSUP, MN_EIDINVAL);
+
+   CEPollDesc& d = p->second;
+
+   d.clearAll();
+
+   return 0;
+}
+
+
+void CEPoll::clear_ready_usocks(CEPollDesc& d, int direction)
```

Comment:
This is necessary for a situation when a socket was tracked for some event, but it no longer needs tracking after the event has happened. Usually that's the case for a connection-pending socket.